### ISSUE TITLE
Slashes disappear when searching

### ISSUE THREAD

**Issue by mxmou** [LLK/scratch-www#5486]
When searching for a string containing a forward slash, the slash and anything after it disappears. For example, entering "scratch/project" searches for just "scratch".

### Steps to Reproduce

1. Enter "scratch/project" or any other string containing a slash into the search bar
2. Press the enter key or click the search button

### Operating System and Browser

Edge 90 on Windows

**Comment by AmazingMech2418** [comment]
Can reproduce. It seems like it's likely meant as a URL sanitation thing for the search API. 

If you URL encode the "/" twice, it works. https://scratch.mit.edu/search/projects?q=scratch%252Fprojects

@BryceLTaylor I think the issue is here:
https://github.com/LLK/scratch-www/blob/32dea2d2a823756882730042b23820718a684443/src/views/search/search.jsx#L62
Here, it runs `decodeURIComponent` before it does it again here:
https://github.com/LLK/scratch-www/blob/32dea2d2a823756882730042b23820718a684443/src/views/search/search.jsx#L84

Only the second one is needed in this case. 

However, since it decodes twice, and encodes it again when searching, only one `decodeURIComponent` is needed.

Here is where it encodes again:
https://github.com/LLK/scratch-www/blob/32dea2d2a823756882730042b23820718a684443/src/views/search/search.jsx#L99

**Comment by AmazingMech2418** [comment]
I would be willing to submit a PR if that would help. It's a really simple fix, just removing the first `decodeURIComponent`.

**Comment by mxmou** [comment]
> I would be willing to submit a PR if that would help. It's a really simple fix, just removing the first `decodeURIComponent`.

Does #5221 fix this?

**Comment by AmazingMech2418** [comment]
> > I would be willing to submit a PR if that would help. It's a really simple fix, just removing the first `decodeURIComponent`.
> 
> Does #5221 fix this?

It does not seem to. The two `decodeURIComponent`s are still there.

**Comment by benjiwheeler** [comment]
@AmazingMech2418 @mxmou I agree that 5221 doesn't fix the problem. @AmazingMech2418, I'd prefer to see a modification of 5221 that would solve the problem using that approach.

**Comment by BryceLTaylor** [comment]
It looks like any work done to fix this issue needs to come along with a number of automated tests to make sure we're not breaking desired behavior.

**Comment by AmazingMech2418** [comment]
@benjiwheeler I'm sorry, I think I must have missed your response somehow. I think 5221 and this can both be merged without conflicts.
